Add a blog-post archetype and blogging documentation

cnunciato opened the pull request:
This change adds documentation for creating and publishing blog posts, along with a [Hugo archetype](https://gohugo.io/content-management/archetypes/) to make generating new blog posts easier:

```
hugo new --kind blog-post blog/my-new-post
```

Fixes #1370.

cnunciato commented on archetypes/blog-post/index.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "{{ replace .Name "-" " " | title }}"
+date: {{ .Date }}
+publishDate: {{ .Date }}
```
I agree, yeah. I've updated the doc and will remove it.

justinvp commented on archetypes/blog-post/index.md:
```diff
@@ -0,0 +1,42 @@
+date: {{ .Date }}
+publishDate: {{ .Date }}
+draft: true
+meta_image: meta.png
```
And maybe another test that ensures the placeholder image doesn't exist at all for the post, to ensure it's been deleted.

Perhaps some of these test would only be run if `hugo.Environment` is not `development`, so the build wouldn't fail during development when you're just getting started writing the post and haven't swapped in a new image yet, but would fail in the environments we want them to fail like staging and production.

cnunciato replied:
I do like that, yeah. Good suggestions, I'll adjust. 💤

aydrian replied:
Is there a way to test if the image is the correct size? They should be 2:1 in dimension and we've been doing 1200x600.

cnunciato commented on archetypes/blog-post/index.md:
```diff
@@ -0,0 +1,42 @@
+draft: true
+meta_image: meta.png
+authors:
+    - joe-duffy
```
Indeed. I just thought it'd be slightly nicer if the first thing you saw wasn't an error, and that we'd catch author mismatches and such at review-time.

justinvp commented on archetypes/blog-post/index.md:
```diff
@@ -0,0 +1,41 @@
+---
+title: "{{ replace .Name "-" " " | title }}"
+date: {{ .Date }}
```
What format does the date end up looking like? Should we do something like `{{ .Date.Format "2006-01-02" }}` so it ends up being simpler like 2019-07-17?

cnunciato replied:
It ends up a `datetime` (with timezone offset), which is actually a good thing IMO, since if you do want to schedule a post to go out after a particular date and time, it's clear how to express that.

cnunciato commented:
@justinvp what do you think would be the best home for the how-to doc? Since we can't really place it at  `content/blog/README.md`, and it doesn't quite belong in the top-level README, would the wiki be a reasonable choice? Or should we make a new file for it specifically somewhere? I'd prefer it were publicly accessible, so the wiki seemed like it'd work, but I also might be overlooking a better place for it.

justinvp commented:
> what do you think would be the best home for the how-to doc?

I don't have a strong opinion. The wiki seems fine to me (and has a lower barrier to edit), though not sure how discoverable it is vs. a file in the repo. I'd also be fine with a top-level `BLOG-GUIDE.md` or `BLOG-HOW-TO.md` or `HOW-TO-BLOG.md` (or better name).

cnunciato commented on archetypes/blog-post/index.md:
```diff
@@ -0,0 +1,41 @@
+
+## Images
+
+![Placeholder Image](meta.png)
```
No it'll work, because an actual placeholder image gets bundled with the archetype.

cnunciato commented:
@justinvp @chrsmith PTAL. I've incorporated most feedback, placed the BLOGGING.md doc at the root of the repo, added cross-references between it and CONTRIBUTING.md, and added field-level comments to the archetype template. 

I chose to stop short of changing the way the author and tag checks behave, as I think with the docs and the commented fields we have now, and the placeholder values, everything should be clear.

justinvp commented on BLOGGING.md:
````diff
@@ -0,0 +1,141 @@
+    hugo new --kind blog-post blog/my-new-post
+    ```
+
+    This will create a minimal post browseable at http://localhost:1313/blog/. You'll find the new post's source file at `content/blog/my-new-post/_index.md` containing the set of [Hugo front matter](https://gohugo.io/content-management/front-matter/) properties you'll need to get started:
````
```suggestion
    This will create a minimal post browsable at http://localhost:1313/blog/. You'll find the new post's source file at `content/blog/my-new-post/_index.md` containing the set of [Hugo front matter](https://gohugo.io/content-management/front-matter/) properties you'll need to get started:
```

justinvp commented on BLOGGING.md:
```diff
@@ -0,0 +1,141 @@
+------------- | ------------- | ------- | ------------------------
+2:1           | 1600×800      | PNG     | Opaque (No Transparency)
+
+Remember to replace the `meta_image` placeholder (or remove the property altogether) before submitting your post.
```
```suggestion
Remember to replace the `meta_image` placeholder (or remove the property altogether and delete the placeholder `meta.png` file) before submitting your post.
```

cnunciato commented:
Actually, my changes to CONTRIBUTING seem to have been left out here. Adding...
